api/controller/contestants.go

```diff
@@ -36,23 +36,25 @@ func GetSingleContestant(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	contestants, err := fetchContestansFromCSV(resp, w)
+	contestants, err := fetchContestansList(resp, w)
 	if err != nil {
 		fmt.Println(err)
```

Comment:
Return an error to the user here as well, even if the following for won't execute with an empty data set, it's better to just return whenever something bad happened and not continue. So I guess you would end up moving the 404 response up here